### ISSUE TITLE
Map Editor

### ISSUE THREAD

**Issue by lumbar527** [Dungeons-of-Kathallion/Dungeons-of-Kathallion#32]
**Is your feature request related to a problem? Please describe.**
It is difficult to design the map if you are just using text files.

**Describe the solution you'd like**
I would like there to be a map editor where you can draw locations and add stuff to them.

**Describe alternatives you've considered**
None. Suggestions of alternatives welcome.

**Additional context**

**Comment by OcelotWalrus** [comment]
Nice! I see you're going into this game back! Right when I go back to Bane Of Wargs. Bane Of Wargs as a simple yet but useful map editor: https://github.com/Dungeons-of-Kathallion/Bane-Of-Wargs-Map-Editor/blob/master/map.py (I'm working on my the script more optimized as many stuff can be simplified); I'm not sure Bane Of Warg's map editor is what you're looking for but at least you're aware of it.

**Comment by lumbar527** [comment]
Yeah, I know about your map editor. I was thinking about something that looked more like drawing. I might be able to put something together in Unity at some point.

What are the programming languages/engines you use? Just wondering about collaborating more on something in the future.

**Comment by OcelotWalrus** [comment]
I can easily code in Python, Lua and C++ (C#) a bit. But you can really just give me documentation, so tutorial and I can code in anything you want. All programming languages really have the same pattern.

About programming engines, I don't really have experience in any :|

> I was thinking about something that looked more like drawing

I really wonder how you're imagining that editor based on drawing.

**Comment by lumbar527** [comment]
> I really wonder how you're imagining that editor based on drawing.

Like, for making new map points, you move the cursor over a grid. The blocked off directions are created automatically and then you can interact with a single room to place stuff.

**Comment by OcelotWalrus** [comment]
> > I really wonder how you're imagining that editor based on drawing.
> 
> Like, for making new map points, you move the cursor over a grid. The blocked off directions are created automatically and then you can interact with a single room to place stuff.
> 
> Any experience with [Pico-8](https://www.lexaloffle.com/pico-8.php)?

Ooh! I see! It'd be nice indeed. I'll look into that Pico-8 thing.

**Comment by OcelotWalrus** [comment]
So you want some kind of tilemap editor, where you could click on each tile and define the point's properties (zone, dialog), and where coordinates, blocked directions and point names would be done automatically.

**Comment by lumbar527** [comment]
Yeah, pretty much.

<details><summary></summary>
<p>

Pico-8 can be used online for free at https://www.pico-8-edu.com/.

</p>
</details>

**Comment by OcelotWalrus** [comment]
I'll check how it works then. And see how is the file it generates. To see if it can be adapted in any way.

**Comment by OcelotWalrus** [comment]
I don't see how you save the thing.

**Comment by lumbar527** [comment]
> I don't see how you save the thing.

The Pico-8 online things? Type `save name` in the console (if in another view hit `esc` to access) and it will download the file. The blue bunny in the corner can offer some help if you click on it as well.

**Comment by OcelotWalrus** [comment]
After searching, I think you'd have to create your own.

**Comment by lumbar527** [comment]
Here's a low-res mockup of what it could look like:
![Editor Mockup](https://github.com/user-attachments/assets/86939efb-1db7-475a-bd9b-99874b5cffa5)
You can select different tools like draw (the crude cursor) to make the points, add item (crude sword) where you will either select a tile to add items to or an item to add to tiles. Same with the monster. There should also be a dialog tool where you select a tile and type in the dialog.

**Comment by lumbar527** [comment]
It should be in higher res than the mockup image is.
I'll try putting some actual programming together soon.

**Comment by lumbar527** [comment]
I'm actually starting to work on an in-game editor, which is both easier (as it's just text) and could be interesting.

**Comment by OcelotWalrus** [comment]
> I'm actually starting to work on an in-game editor, which is both easier (as it's just text) and could be interesting.

Oh nice! Can't wait to see it!
